palnning: open space smoother jacobian parrallel implementation

deidaraho opened the pull request.

Capri2014 commented:
@deidaraho better add test to ensure that you do not fat figure when hand writing Jacobian.

deidaraho commented:
> > @deidaraho better add test to ensure that you do not fat figure when hand writing Jacobian.
> 
> I added a flag to enable derivative checking inside ipopt for debugging jacobians.

I have modified the ipopt's test to check gradient with parallel implementation.

jinghaomiao commented:
LGTM
